web/src/scroll_bar.ts

```diff
@@ -9,11 +9,17 @@ import {user_settings} from "./user_settings";
 // number of element widths based on the value detected here.
 //
 // From https://stackoverflow.com/questions/13382516/getting-scroll-bar-width-using-javascript
-function getScrollbarWidth() {
+function getScrollbarWidth(): number {
     const outer = document.createElement("div");
     outer.style.visibility = "hidden";
     outer.style.width = "100px";
-    outer.style.msOverflowStyle = "scrollbar"; // needed for WinJS apps
+
+    /*
+    `-ms-overflow-style` is a vendor specific property. Therefore,
+    `.setProperty(name, value)` syntax is used to make typescript
+    happy. This property is needed for WinJS apps.
+    */
+    outer.style.setProperty("-ms-overflow-style", "scrollbar");
```

Comment:
WinJS is not a thing anymore; we can delete this.

Reply:
Got it. Btw the current error is a really niche one, look at this:-
```ts
error  'set_layout_width' was used before it was defined  @typescript-eslint/no-use-before-define
```
The functions will be hoisted to the top of this module's scope so at runtime this error wouldn't be possible. This really is just typescript-eslint being extra strict. Should I fix this? Or are we gonna change the rules to ignore this error?

Reply:
@andersk Fixed it.